src/ppx/xr_component.cpp

```diff
@@ -856,40 +848,72 @@ glm::mat4 XrComponent::GetProjectionMatrixForCurrentViewAndSetFrustumPlanes(floa
     mNearPlaneForFrame = nearZ;
     mFarPlaneForFrame  = farZ;
 
-    const float tan_left  = tanf(fov.angleLeft);
-    const float tan_right = tanf(fov.angleRight);
-
-    const float tan_down = tanf(fov.angleDown);
-    const float tan_up   = tanf(fov.angleUp);
-
-    const float tan_width  = tan_right - tan_left;
-    const float tan_height = (tan_up - tan_down);
-
-    const float a00 = 2 / tan_width;
-    const float a11 = 2 / tan_height;
-
-    const float a20 = (tan_right + tan_left) / tan_width;
-    const float a21 = (tan_up + tan_down) / tan_height;
-    const float a22 = -farZ / (farZ - nearZ);
+    for (XrCamera& camera : mCameras) {
+        camera.SetFrustumPlanes(nearZ, farZ);
+    }
+}
 
-    const float a32 = -(farZ * nearZ) / (farZ - nearZ);
+// -------------------------------------------------------------------------------------------------
+// XrCamera
+// -------------------------------------------------------------------------------------------------
 
-    // clang-format off
-    const float mat[16] = {
-        a00,    0,      0,      0,
-        0,      a11,    0,      0,
-        a20,    a21,    a22,    -1,
-        0,      0,      a32,    0,
-    };
-    // clang-format on
+void XrCamera::UpdateView(const XrView& view)
+{
+    mView = view;
+    XrCamera::UpdateCamera();
+}
 
-    return glm::make_mat4(mat);
+void XrCamera::SetFrustumPlanes(float nearZ, float farZ)
+{
+    if (mNearClip == nearZ && mFarClip == farZ) {
+        return;
+    }
+    mNearClip = nearZ;
+    mFarClip  = farZ;
+    UpdateCamera();
 }
 
-XrPosef XrComponent::GetPoseForCurrentView() const
+void XrCamera::UpdateCamera()
 {
-    PPX_ASSERT_MSG((mCurrentViewIndex < mViews.size()), "Invalid view index!");
-    return mViews[mCurrentViewIndex].pose;
+    const float3 forward = PPX_CAMERA_DEFAULT_VIEW_DIRECTION;
+    const float3 up      = PPX_CAMERA_DEFAULT_WORLD_UP;
+    // Given:
+    //   - mPixelAligned
+    //   - mNearClip, mFarClip
+    //   - view.fov.{angleLeft,angleRight,angleDown,angleUp}
+    //   - view.pose.{orientation,position}
+    // Copied:
+    //   - mEyePosition = view.pose.position
+    // Solve for:
+    //   - mViewDirection
+    //   - mTarget
+    //   - mWorldUp
+    //   - all matrices (using LookAt)
+
+    // Calculate projection matrix
+    {
+        const XrFovf& fov = mView.fov;
+
+        const float left  = tanf(fov.angleLeft) * mNearClip;
+        const float right = tanf(fov.angleRight) * mNearClip;
+
+        const float down = tanf(fov.angleDown) * mNearClip;
+        const float up   = tanf(fov.angleUp) * mNearClip;
+
+        // Note: the current variant in use is frustumRH_ZO.
```

Comment:
super-duper-nit: this kind of comment is bound to become invalid at some point since changing global constants is done somewhere else.

Reply:
It's somewhat intentional, I wrote down this comment so in case something changed in the future and we discover xr need to stay on the same  `glm::frustum` variant, we know which one it is.